typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -155,7 +175,7 @@ export class MultiProvider<MetaExt = {}> extends ChainMetadataManager<MetaExt> {
     if (signer.provider) return signer;
     // Auto-connect the signer for convenience
     const provider = this.tryGetProvider(chainName);
-    return provider ? signer.connect(provider) : signer;
+    return provider ? signer.connect(provider as any) : signer;
```

Comment:
does this mean that every consumer of the multi provider has to do this type narrowing of `getProvider`?

Reply:
So the quickest solution to this is handle zksync provider separately, but it doesn't really scale: 

```
    if (signer instanceof ZKSyncWallet) {
      return signer.connect(provider as ZKSyncProvider);
    }
    return (signer as EthersSigner).connect(provider);
```

Reply:
@jmrossy 's comment helped me resolve this. Now typecasting is no longer needed for `signer.connect()`